dask/dataframe/tests/test_dataframe.py

```diff
@@ -3320,3 +3320,11 @@ def test_partitions_indexer():
     assert ddf.x.partitions[:3].npartitions == 3
 
     assert ddf.x.partitions[::2].compute().tolist() == [0, 1, 4, 5, 8, 9]
+
+
+def test_setitem():
+    df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
+    ddf = dd.from_pandas(df, 2)
```

Comment:
Because we're doing inplace operations we should probably defensively call `df.copy()` here just to make sure we're not having thing work accidentally.

Reply:
Good catch! I fixed it.